build.gradle.kts

```diff
@@ -0,0 +1,234 @@
+
+import edu.wpi.first.wpilib.versioning.ReleaseType
+import org.gradle.api.Project
+//import org.gradle.api.plugins.quality.FindBugs
+import org.gradle.api.tasks.wrapper.Wrapper
+import org.gradle.testing.jacoco.tasks.JacocoReport
+
+buildscript {
+    repositories {
+        mavenCentral()
+    }
+    dependencies {
+        classpath("org.junit.platform:junit-platform-gradle-plugin:1.0.0")
+    }
+}
+plugins {
+    application
+    `maven-publish`
+    jacoco
+    java
+    //checkstyle
+    //pmd
+    id("edu.wpi.first.wpilib.versioning.WPILibVersioningPlugin") version "1.6"
+    id("com.github.johnrengelman.shadow") version "2.0.1"
+    id("com.diffplug.gradle.spotless") version "3.5.1"
+}
+
+apply {
+    //plugin("pmd")
+    //plugin("findbugs")
+    plugin("jacoco")
+    plugin("org.junit.platform.gradle.plugin")
+}
+
+group = "edu.wpi.first.wpilib"
+
+// Spotless is used to lint and reformat source files.
+spotless {
+    kotlinGradle {
+        // Configure the formatting of the Gradle Kotlin DSL files (*.gradle.kts)
+        ktlint("0.9.1")
+        endWithNewline()
+    }
+}
+
+repositories {
+    mavenCentral()
+    maven { setUrl("https://jitpack.io") }
+}
+
+dependencies {
+    compile(group = "org.apache.velocity", name = "velocity", version = "1.7")
+    compile(group = "org.yaml", name = "snakeyaml", version = "1.10")
+    compile(group = "com.github.jgraph", name = "jgraphx", version = "v3.3.1.1")
+    compile(group = "org.projectlombok", name = "lombok", version = "1.16.16")
+
+    fun junitJupiter(name: String, version: String = "5.0.0") =
+            create(group = "org.junit.jupiter", name = name, version = version)
+
+    testCompile(junitJupiter(name = "junit-jupiter-api"))
+    testCompile(junitJupiter(name = "junit-jupiter-engine"))
+    testCompile(junitJupiter(name = "junit-jupiter-params"))
+    testCompile(group = "junit", name = "junit", version="4.11")
+}
+
+application {
+    mainClassName = "robotbuilder.RobotBuilder"
+}
+
+/*
+checkstyle {
+    configFile = file("$rootDir/checkstyle.xml")
+    toolVersion = "8.1"
+}
+
+pmd {
+    isConsoleOutput = true
+    sourceSets = setOf(java.sourceSets["main"], java.sourceSets["test"])
+    reportsDir = file("${project.buildDir}/reports/pmd")
+    ruleSetFiles = files(file("$rootDir/pmd-ruleset.xml"))
+    ruleSets = emptyList()
+}
+
+findbugs {
+    sourceSets = setOf(java.sourceSets["main"], java.sourceSets["test"])
+    effort = "max"
+}
+
+tasks.withType<FindBugs> {
+    reports {
+        xml.isEnabled = false
+        emacs.isEnabled = true
+    }
+    finalizedBy(task("${name}Report") {
+        mustRunAfter(this@withType)
+        doLast {
+            this@withType
+                    .reports
+                    .emacs
+                    .destination
+                    .takeIf { it.exists() }
+                    ?.readText()
+                    .takeIf { !it.isNullOrBlank() }
+                    ?.also { logger.warn(it) }
+        }
+    })
+}
+*/
```

Comment:
What's wrong with this giant commented out chunk?

Reply:
build fails spectacularly because the code doesn't pass checkstyle or pmd